From 1e084f428b3ccd43d00e7e5954c4440ce48ce2da Mon Sep 17 00:00:00 2001
From: Paul Kocialkowski <contact@paulk.fr>
Date: Sun, 3 Dec 2017 00:08:32 +0100
Subject: [PATCH 4/4] arm64: Target aarch64elf emulation to allow bare-metal
 toolchains

This sets the LDFLAGS variable to target the aarch64elf emulation
instead of aarch64linux, which is incompatible with bare-metal
toolchains.

This change allows the kernel to build with bare-metal toolchains again.

Fixes: 3d6a7b99e3fa ("arm64: ensure the kernel is compiled for LP64")

Cc: stable@vger.kernel.org
Signed-off-by: Paul Kocialkowski <contact@paulk.fr>
---
 arch/arm64/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index 3233261aa031..3e474991a3d4 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -43,12 +43,12 @@ ifeq ($(CONFIG_CPU_BIG_ENDIAN), y)
 KBUILD_CPPFLAGS	+= -mbig-endian
 AS		+= -EB
 LD		+= -EB
-LDFLAGS		+= -maarch64linuxb
+LDFLAGS		+= -maarch64elfb
 else
 KBUILD_CPPFLAGS	+= -mlittle-endian
 AS		+= -EL
 LD		+= -EL
-LDFLAGS		+= -maarch64linux
+LDFLAGS		+= -maarch64elf
 endif
 
 CHECKFLAGS	+= -D__aarch64__
-- 
2.15.0

